iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -388,21 +388,13 @@ public static PartitionData toPartitionData(StructLike sourceKey, Types.StructTy
       Types.StructType targetKeyType) {
     PartitionData data = new PartitionData(targetKeyType);
     for (int i = 0; i < targetKeyType.fields().size(); i++) {
-
-      int fi = i;
-      String fieldName = targetKeyType.fields().get(fi).name();
-      Object val = sourceKeyType.fields().stream()
-          .filter(f -> f.name().equals(fieldName)).findFirst()
-          .map(sourceKeyElem -> sourceKey.get(sourceKeyType.fields().indexOf(sourceKeyElem),
-              targetKeyType.fields().get(fi).type().typeId().javaClass()))
-          .orElseThrow(() -> new RuntimeException(
-              String.format("Error retrieving value of partition field %s", fieldName)));
-
-      if (val != null) {
-        data.set(fi, val);
-      } else {
-        throw new RuntimeException(String.format("Partition field's %s value is null", fieldName));
-      }
+      String fieldName = targetKeyType.fields().get(i).name();
```

Comment:
Could we simplify this with
````
StructProjection projection =
    StructProjection.createAllowMissing(sourceKeyType, targetKeyType);
projection.wrap(sourceKey);
  
PartitionData keyTemplate = new PartitionData(targetKeyType);
keyTemplate.copyFor(sourceKey);
````

Reply:
above `PartitionData toPartitionData(StructLike key, Types.StructType keyType)` could be replaced with 
````
PartitionData keyTemplate = new PartitionData(keyType);
return keyTemplate.copyFor(key);
````

Reply:
since you are doing refactor here, could we also leverage `IcebergTableUtil#toPartitionData` in `HiveIcebergStorageHandler#getPartishSummary`? see

````
        try (Closeable toClose = partitionStatsRecords) {
          PartitionStats partitionStats = Iterables.tryFind(partitionStatsRecords, stats -> {
            PartitionSpec spec = table.specs().get(stats.specId());
            
            PartitionData data  = IcebergTableUtil.toPartitionData(stats.partition(), 
                partitionType, spec.partitionType());
            return spec.partitionToPath(data).equals(partish.getPartition().getName());

          }).orNull();
````

Reply:
hi @okumin, WDYT, does it make sense?

Reply:
@deniskuzZ Thanks. I found HIVE-28727 has been resolved on the master branch, maybe thanks to [HIVE-28581](https://github.com/apache/hive/commit/dee65466b8dcb0a624e65f689e9580425deb34ab)! I am thinking to void this issue

Reply:
I updated two files based on your suggestion. I will update the description of the JIRA ticket and this PR if CI succeeds